test/unit.js

```diff
@@ -1,16 +1,16 @@
 'use strict';
+require('babel-polyfill');
 const Promise = require('bluebird');
 const expect = require('chai').expect;
 let sailsRequest;
 
 describe('unit tests', () => {
-  before(function () {
+  before(async function () {
     this.timeout(5000);
-    return Promise.promisify(require('sails').load)().then(() => {
-      /* sails.request takes a callback instead of returning a Promise, so Promisify it and use a new `sailsRequest` function,
-      which is exactly the same except that it returns a Promise. */
-      sailsRequest = Promise.promisify(sails.request);
-    });
+    await Promise.promisify(require('sails').load)();
+    /* sails.request takes a callback instead of returning a Promise, so Promisify it and use a new `sailsRequest` function,
+    which is exactly the same except that it returns a Promise. */
+    sailsRequest = Promise.promisify(sails.request);
   });
   it('returns a 404 error when fetching a nonexistent path', () => {
     return sailsRequest({method: 'get', url: '/this-path-is-invalid'}).catch(err => {
```

Comment:
A try/catch would be a bit tricky because we would want to call expect.fail() if the request succeeds without errors, but the expect.fail() error itself would be caught by the catch block. But I fixed it so it should fail if the request succeeds.